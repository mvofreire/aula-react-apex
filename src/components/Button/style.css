.btn {
  padding: 10px;
  border: none;
  outline: none;
  border-radius: 2px;
  cursor: pointer;
}

.btn.primary {
  background: cornflowerblue;
  color: #fff;
  text-shadow: 1px 1px 1px #000;
}
.btn.primary:hover {
  background: #3d60a0;
  color: #fff;
  text-shadow: 1px 1px 1px #000;
}

.btn.secondary {
  background: #bd4646;
  color: #fff;
  text-shadow: 0px 0px 3px #000;
}
.btn.secondary:hover {
  background: #8b1c1c;
  color: #fff;
  text-shadow: 0px 0px 3px #000;
}
