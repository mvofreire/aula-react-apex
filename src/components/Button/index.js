import React from 'react'
import "./style.css"

function Button({label, onClick}){
  return (
    <button className="btn" onClick={onClick}>
      {label}
    </button>
  )
}

export default Button