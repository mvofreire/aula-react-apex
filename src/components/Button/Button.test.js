import { render, screen, fireEvent, act } from '@testing-library/react';
import Button from './index';

describe('Button', () => {
  test('Should renders in screen', () => {
    render(<Button label="banana" />);
    const buttonLabel = screen.getByText(/banana/i);
    expect(buttonLabel).toBeInTheDocument();
  });

  test('Should Call callback onClick', () => {
    const callback = jest.fn();
    const { getByText } = render(<Button label='button' onClick={callback} />)

    act(() => {
      fireEvent.click(getByText(/button/))
    })

    expect(callback).toBeCalledTimes(1)
  })

  test('Should match with snapshot', () => {
    const { container } = render(<Button label='button' />)
    expect(container).toMatchSnapshot('default')
  })

  test('Should match with snapshot secondary', () => {
    const { container } = render(<Button label='button' type='secondary' />)
    expect(container).toMatchSnapshot('secondary')
  })
})
