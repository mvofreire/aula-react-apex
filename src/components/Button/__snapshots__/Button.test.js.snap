// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match with snapshot secondary: secondary 1`] = `
<div>
  <button
    class="btn secondary "
    type="button"
  >
    button
  </button>
</div>
`;

exports[`Button Should match with snapshot: default 1`] = `
<div>
  <button
    class="btn primary "
    type="button"
  >
    button
  </button>
</div>
`;
